app/src/main/res/layout/activity_git_clone.xml

```diff
@@ -104,75 +150,72 @@
-        app:layout_constraintStart_toEndOf="@id/label_connection_mode" />
-
-    <com.google.android.material.button.MaterialButton
-        style="@style/Widget.MaterialComponents.Button"
-        android:id="@+id/clone_button"
-        android:text="@string/clone_button"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:onClick="cloneRepository"
-        android:textColor="?android:attr/windowBackground"
-        android:layout_marginTop="8dp"
-        app:backgroundTint="?attr/colorSecondary"
+        android:layout_margin="8dp"
         app:layout_constraintTop_toBottomOf="@id/label_connection_mode"
-        app:layout_constraintStart_toStartOf="parent" />
+        app:layout_constraintStart_toStartOf="parent"
+        app:selectionRequired="true"
+        app:singleSelection="true" >
+
+        <com.google.android.material.button.MaterialButton
+            style="?attr/materialButtonOutlinedStyle"
+            android:id="@+id/connection_mode_ssh"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="@string/connection_mode_ssh_key"
+            android:textColor="?android:attr/textColorPrimary"
+            app:rippleColor="@color/ripple_color"
+            app:strokeColor="?attr/colorSecondary"
+            app:backgroundTint="@color/toggle_button_selector" />
+
+        <com.google.android.material.button.MaterialButton
+            style="?attr/materialButtonOutlinedStyle"
+            android:id="@+id/connection_mode_username"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="@string/connection_mode_basic_authentication"
+            android:textColor="?android:attr/textColorPrimary"
+            app:rippleColor="@color/ripple_color"
+            app:strokeColor="?attr/colorSecondary"
+            app:backgroundTint="@color/toggle_button_selector" />
+
+        <com.google.android.material.button.MaterialButton
+            style="?attr/materialButtonOutlinedStyle"
+            android:id="@+id/connection_mode_openkeychain"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:text="@string/connection_mode_openkeychain"
+            android:textColor="?android:attr/textColorPrimary"
+            app:rippleColor="@color/ripple_color"
+            app:strokeColor="?attr/colorSecondary"
+            app:backgroundTint="@color/toggle_button_selector" />
+
+    </com.google.android.material.button.MaterialButtonToggleGroup>
 
     <com.google.android.material.button.MaterialButton
         style="@style/Widget.MaterialComponents.Button"
         android:id="@+id/save_button"
         android:text="@string/crypto_save"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:onClick="saveConfiguration"
```

Comment:
I would like it if the save button would both save the configuration and quit the activity.

Reply:
Fixed with 5baa83765a80fa9cde31fbc9c8e718927b989933